### PR TITLE
Added devices config handling and device HostConfig handling

### PR DESCRIPTION
**Pull request by DanElbert**
I'm new to contributing to compose, but this is a stab at adding the device option to the yml config, as per issue #754.

### CODE REVIEW

**Comment by dnephin** [comment]
I think this is a good start. I believe we also need to update `compose/config.py` to be able to merge the lists of devices when using `extends`

**Comment by DanElbert** [comment]
I've updated the pull request.

File: docs/yml.md [hunk 1]
```diff
@@ -342,6 +342,15 @@ dns_search:
   - dc2.example.com
 ```
 
+### devices
+
+Devices option.  List of device mappings.  Uses the same format as the --device docker client create option.
```
Comment (aanand) [inline docs/yml.md]: - First sentence is unnecessary
- `--device` should be in backticks
- Wrap to 80 chars please!

**Comment by aanand** [comment]
Thanks!

For the sake of consistency, we should support a mapping as well as a list. Is that supported? If so, it should be documented.

**Comment by aanand** [comment]
> For the sake of consistency, we should support a mapping as well as a list.

I retract this, as I was thinking about environment variables. We don't support YAML mappings for volumes. Sorry for the noise.

This looks good - just needs squashing to a single commit.

**Comment by DanElbert** [comment]
Thanks for the feedback.  I was just going to ask about the mapping. :)

**Comment by DanElbert** [comment]
All squashed.  Let me know if I missed anything else.

**Comment by aanand** [comment]
LGTM

**Comment by dnephin** [comment]
LGTM

**Pull request merged**